include/boost/geometry/algorithms/detail/overlay/append_no_dups_or_spikes.hpp

```diff
@@ -99,7 +100,16 @@ inline void append_no_dups_or_spikes(Range& range, Point const& point,
         return;
     }
 
-    traits::push_back<Range>::apply(range, point);
+    auto append = [](auto& r, const auto& p)
+    {
+        using point_t = typename boost::range_value<Range>::type;
+        point_t rp;
+        geometry::detail::conversion::convert_point_to_point(p, rp);
+        traits::push_back<Range>::apply(r, rp);
```

Comment:
The point could be moved here. In practice this wouldn't make any difference but if someone used e.g. Multiprecision coordinates we'd avoid copying.

Reply:
:heavy_check_mark: